signing-scheme.md

```diff
@@ -0,0 +1,73 @@
+
+`notary.default.x509`
+
+* Uses trusts store types Certificate Authority (CA) and Timestamping Authority (TSA) during signature verification. The signature is verified against the trust store of type CA, and the *Timestamp signature* is verified against the trust store of type TSA (to determine the signing time).
+* For signature verification to be successful
+  * The verifying entity’s trust store MUST contain the trusted root certificates under named trust stores of type CA (`{CONFIG}/notation/truststore/x509/ca`) and TSA(`{CONFIG}/notation/truststore/x509/tsa`)
+  * The named trust stores MUST be specified in *trustpolicy.json*. E.g. *trustPolicy.trustStore* with value of `ca:acme-rockets,tsa:acme-tsa`.
+
+`notary.signingAuthority.x509`
+
+* Uses trusts store type Signing Authority during signature verification. The signature is verified against the trust store of type Signing Authority. The signing time is determined using the *Authentic Signing time* attribute in the signature envelope, and does not rely on a seperate TSA generated Timestamp signature.
+* For signature verification to be successful
+  * The verifying entity’s trust store MUST contain the trusted root certificates under named trust stores of type CA (`{CONFIG}/notation/truststore/x509/signingAuthority`).
+  * The named trust stores MUST be specified in *trustpolicy.json*. E.g. *trustPolicy.trustStore* with value of `signingAuthority:foobar` .
+
+## FAQ
+
+*Q:* What is the relationship of Signing Scheme with Signature Envelope format?
+*A:* Signing Scheme aims to be agnostic of the Signature Envelope format. A given signing scheme can be implemented through any signature envelope format (such as JWS or COSE) as long as it can support the required signature schema used by the signing scheme.
+
+*Q:* Why trust store for Signing Authority (`x509/signingAuthority`) is distinct from trust store for Certificate Authority (`x509/ca`), why can’t they share the same trust store?
+*A:* Signing Authority is a different type of trusted entity as compared to Certificate Authority (CA) or Timestamping Authority (TSA) which are much widely used. A CA is trusted for verifying the identity of a signing entity (end user) and issuing it a certificate, whereas a TSA is trusted to generate authentic timestamp. In contrast, an SA is trusted for generating signatures on behalf of a signing entity (end user) and also to generate authentic timestamp. If we use a shared trust store for CA and SA roots, an end user can self generate a signature with signing scheme `notary.signingAuthority.x509` and misrepresent themselves as a signing authority that would have ideally generated the signature. A verifying entity places certain trust in a signing authority - like the ability to generate authentic timestamp, and have uniform policies related to signature generation process, which aren't available when an end user generates a signature themselves.
```

Comment:
As discussed we should add some blurb around `separating trust store gives verifier control to dictate what they trust as CA and signing authority`.

Reply:
Made some changes in the wording, let me know if you have further feedback.